core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -113,7 +115,8 @@ public IndexService(IndexSettings indexSettings, NodeEnvironment nodeEnv,
                         IndexStore indexStore,
                         IndexEventListener eventListener,
                         IndexModule.IndexSearcherWrapperFactory wrapperFactory,
-                        MapperRegistry mapperRegistry) throws IOException {
+                        MapperRegistry mapperRegistry,
+                        IndexingMemoryController indexingMemoryController) throws IOException {
```

Comment:
Can this be a `IndexingOperationListener... listeners` that way we don't introduce a hard dependency on IMC